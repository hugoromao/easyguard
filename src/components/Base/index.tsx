"use client";

import React, { useContext } from "react";

import { Inter } from "next/font/google";

import { GlobalContext } from "@/context/global";

import Navbar from "@/components/Navbar";
import NewPasswordForm from "@/components/NewPasswordForm";

const inter = Inter({ subsets: ["latin"] });

type BaseProps = {
  children: React.ReactNode;
};

const Base = ({ children }: BaseProps) => {
  const { isOpen, onOpenChange } = useContext(GlobalContext);

  return (
    <>
      <main
        style={inter.style}
        className="h-[calc(100dvh)] grid grid-rows-[1fr_auto]"
      >
        {children}
        <Navbar />
      </main>

      <NewPasswordForm isOpen={isOpen} onOpenChange={onOpenChange} />
    </>
  );
};

export default Base;
