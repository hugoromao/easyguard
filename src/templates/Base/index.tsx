"use client";

import React, { useState } from "react";

import { Inter } from "next/font/google";

import Navbar from "@/components/Navbar";
import NewPasswordForm from "@/components/NewPasswordForm";

const inter = Inter({ subsets: ["latin"] });

type BaseProps = {
  children: React.ReactNode;
};

const Base = ({ children }: BaseProps) => {
  const [isNewPasswordFormActive, setIsNewPasswordFormActive] = useState(false);

  function openNewPasswordForm() {
    setIsNewPasswordFormActive(true);
  }

  return (
    <>
      <main
        style={inter.style}
        className="h-[calc(100dvh)] grid grid-rows-[1fr_auto]"
      >
        {children}
        <Navbar openNewPasswordForm={openNewPasswordForm} />
      </main>

      <NewPasswordForm
        active={isNewPasswordFormActive}
        setActive={setIsNewPasswordFormActive}
      />
    </>
  );
};

export default Base;
