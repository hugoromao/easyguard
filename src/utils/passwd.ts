import {
  commonPortugueseWords,
  keyboardPatterns,
  leetSpeakLetters,
  specialCaracters,
} from "./patterns";

export function validateUserWords(words: string[]): CustomError[] {
  const errors: CustomError[] = [];

  if (words.length < 3) {
    errors.push({
      type: "wordError",
      message: "Insira no mínimo três palavras.",
      index: undefined,
    });
  }

  words.forEach((word, index) => {
    if (word.length < 3) {
      errors.push({
        type: "wordError",
        message: `A palavra “${word}” é menor que três caracteres.`,
        index,
      });
    }
  });

  const wordsSet = new Set(words);
  if (words.length !== wordsSet.size) {
    errors.push({
      type: "wordError",
      message: "Existem palavras repetidas.",
      index: undefined,
    });
  }

  words.forEach((word, index) => {
    if (commonPortugueseWords.includes(word)) {
      errors.push({
        type: "wordError",
        message: `${word} é uma palavra muito comum`,
        index,
      });
    }
    if (keyboardPatterns.includes(word)) {
      errors.push({
        type: "wordError",
        message: "Não utilize padrões de teclado",
        index,
      });
    }
  });

  return errors;
}

export function validateUserNumbers(numbers: number[]) {
  const errors: CustomError[] = [];

  if (numbers.length < 2) {
    errors.push({
      type: "numberError",
      message: "Insira no mínimo dois números.",
      index: undefined,
    });
  }

  const numbersSet = new Set(numbers);
  if (numbers.length !== numbersSet.size) {
    errors.push({
      type: "numberError",
      message: "Existem números repetidos.",
      index: undefined,
    });
  }

  return errors;
}

export function toLeetSpeak(string: string): string {
  const randomString = string[Math.floor(Math.random() * string.length)] as "A";

  if (leetSpeakLetters[randomString] !== undefined) {
    return string.replaceAll(randomString, leetSpeakLetters[randomString]);
  }

  return string;
}

export function validateEntropy(password: string): number {
  let chars = 0;
  if (/[a-z]/.test(password)) chars = chars + 26;
  if (/[A-Z]/.test(password)) chars = chars + 26;
  if (/[0-9]/.test(password)) chars = chars + 10;
  if (/[!"£$%^&*()]/.test(password)) chars = chars + 10;
  if (/[`¬\-=_+[\]{};'#:@~,./<>?\\|]/.test(password)) chars = chars + 23;

  let entropy = parseInt(
    String(Math.log2(Math.pow(chars, password.length))),
    10
  );

  return entropy;
}

type CustomError = {
  type: "wordError" | "numberError" | "generatedPasswordError";
  index: number | undefined;
  message: string;
};

type Data = {
  success: boolean;
  password: string | null;
  entropy: number;
  errors: CustomError[];
};

export function generatePassword(iw: string[], inu: number[]): Data {
  const words = [...iw];
  const numbers = [...inu];

  let generatedPasswordErrors: CustomError[] = [];
  let wordErrors = validateUserWords(words);
  let numberErrors = validateUserNumbers(numbers);

  let password = null;
  let entropy = 0;

  let hasErrors = !wordErrors.length && !numberErrors.length;

  let temp: string[] = [];

  words.forEach((word, index) => {
    temp.push(word);
    if (index === words.length - 1) return;
    const randomNumber = Math.floor(Math.random() * numbers.length);

    if (randomNumber === 1 || numbers.length === 0) {
      const randomSpecialIndex = Math.floor(
        Math.random() * specialCaracters.length
      );
      temp.push(specialCaracters[randomSpecialIndex]);
    }

    if (randomNumber === 0 && numbers.length !== 0) {
      const randomNumberIndex = Math.floor(Math.random() * numbers.length);
      temp.push(String(numbers[randomNumberIndex]));
      numbers.splice(randomNumberIndex, 1);
    }
  });

  const shouldUseLeetSpeak = Math.ceil(Math.random() * 2);
  password =
    shouldUseLeetSpeak === 1 ? toLeetSpeak(temp.join("")) : temp.join("");

  entropy = validateEntropy(password);

  if (entropy < 60) {
    generatedPasswordErrors.push({
      type: "generatedPasswordError",
      message: "A senha gerada possui entropia menor que 60 bits",
      index: undefined,
    });
  }

  if (password.length < 16) {
    generatedPasswordErrors.push({
      type: "generatedPasswordError",
      message: "A senha gerada é menor que 16 caracteres",
      index: undefined,
    });
  }

  hasErrors =
    !wordErrors.length && !numberErrors.length && !generatePassword.length;

  // TODO: Validate against the following password meters: Passfault, The password meter and How secure is my password.
  return {
    success: !hasErrors,
    password: password,
    entropy,
    errors: [...wordErrors, ...numberErrors, ...generatedPasswordErrors],
  };
}
