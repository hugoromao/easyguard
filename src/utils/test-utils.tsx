import { NextUIProvider } from "@nextui-org/react";
import { render, RenderOptions } from "@testing-library/react";
import {
  AppRouterContext,
  AppRouterInstance,
} from "next/dist/shared/lib/app-router-context.shared-runtime";

import {
  GlobalContext,
  GlobalContextType,
  GlobalContextDefaultValues,
} from "@/context/global";

type CustomRenderProps = {
  globalProviderProps?: GlobalContextType;
} & Omit<RenderOptions, "queries">;

const customRender = (
  ui: React.ReactElement,
  {
    globalProviderProps = GlobalContextDefaultValues,
    ...renderOptions
  }: CustomRenderProps = {},
  router?: Partial<AppRouterInstance>
) => {
  const mockedRouter: AppRouterInstance = {
    back: jest.fn(),
    forward: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    refresh: jest.fn(),
    prefetch: jest.fn(),
    ...router,
  };

  return render(
    <AppRouterContext.Provider value={mockedRouter}>
      <NextUIProvider>
        <GlobalContext.Provider value={globalProviderProps}>
          {ui}
        </GlobalContext.Provider>
      </NextUIProvider>
    </AppRouterContext.Provider>,
    renderOptions
  );
};

export * from "@testing-library/react";
export { customRender as render };
