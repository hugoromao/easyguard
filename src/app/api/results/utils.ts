import { validateEntropy } from "@/utils/passwd";

export const gabarito = {
  qo1: "false",
  qo2: "false",
  qo3: "true",
  qo4: "true",
  qo5: "false",
  qo6: "true",
  qo7: "false",
  qo8: "true",
  qo9: "false",
  qo10: "true",
  qo11: "false",
  qo12: "false",
  qo13: "false",
  qo14: "true",
  qo15: "true",
  qo16: "false",
  qo17: "true",
  qo18: "true",
  qo19: "false",
  qo20: "false",
};

export function mean(values: number[]) {
  return Number(
    (values.reduce((prev, curr) => prev + curr) / values.length).toFixed(2),
  );
}

export function getKnowledgeTestScorePercent(qos: any) {
  const hitsCount = Object.entries(qos).reduce((count, [key, value]) => {
    return count + (gabarito[key as "qo1"] === value ? 1 : 0);
  }, 0);

  return (hitsCount / 20).toLocaleString(undefined, {
    style: "percent",
    minimumFractionDigits: 2,
  });
}

export function mapResult(result: any) {
  return result.map(
    // eslint-disable-next-line no-unused-vars
    ({ id, name, posTest, createdAt, updatedAt, qd1, qd2, ...qos }: any) => ({
      name,
      posTest,
      scorePercent: getKnowledgeTestScorePercent(qos),
      passwordLength: qd1.length,
      passwordTotalCharacterGroups: countCharacterGroups(qd1),
      passwordEntropy: validateEntropy(qd1),
    }),
  );
}

export function countCharacterGroups(str: string) {
  const groups = {
    lowerCase: false,
    upperCase: false,
    digits: false,
    specialChars: false,
  };

  for (const char of str) {
    if (/[a-z]/.test(char)) {
      groups.lowerCase = true;
    } else if (/[A-Z]/.test(char)) {
      groups.upperCase = true;
    } else if (/\d/.test(char)) {
      groups.digits = true;
    } else {
      groups.specialChars = true;
    }
  }

  return Object.values(groups).filter(Boolean).length;
}
