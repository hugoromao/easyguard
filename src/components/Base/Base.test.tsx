import { render, screen } from "@/utils/test-utils";
import Base from ".";

describe("<Base />", () => {
  it("should render correctly", () => {
    const component = render(<Base>test</Base>);
    expect(screen.getByText(/test/i));
    expect(component).toMatchSnapshot();
  });
});
