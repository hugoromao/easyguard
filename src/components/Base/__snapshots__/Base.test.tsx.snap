// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <main
          class="h-[calc(100dvh)] grid grid-rows-[1fr_auto]"
          style="font-family: fontFamily;"
        >
          test
          <nav
            class="flex bg-white p-1 justify-center border-t-2 border-default-200"
          >
            <a
              class="p-4 rounded-xl "
              href="/"
            >
              <svg
                aria-hidden="true"
                class="h-8 text-default-400"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
            <a
              class="p-4 rounded-xl hover:bg-default-100"
              href="/achivements"
            >
              <svg
                aria-hidden="true"
                class="h-8 text-default-400"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 11.25v8.25a1.5 1.5 0 0 1-1.5 1.5H5.25a1.5 1.5 0 0 1-1.5-1.5v-8.25M12 4.875A2.625 2.625 0 1 0 9.375 7.5H12m0-2.625V7.5m0-2.625A2.625 2.625 0 1 1 14.625 7.5H12m0 0V21m-8.625-9.75h18c.621 0 1.125-.504 1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125h-18c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
            <a
              class="p-4 rounded-xl hover:bg-default-100"
              href="/settings"
            >
              <svg
                aria-hidden="true"
                class="h-8 text-default-400"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.325.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 0 1 1.37.49l1.296 2.247a1.125 1.125 0 0 1-.26 1.431l-1.003.827c-.293.241-.438.613-.43.992a7.723 7.723 0 0 1 0 .255c-.008.378.137.75.43.991l1.004.827c.424.35.534.955.26 1.43l-1.298 2.247a1.125 1.125 0 0 1-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.47 6.47 0 0 1-.22.128c-.331.183-.581.495-.644.869l-.213 1.281c-.09.543-.56.94-1.11.94h-2.594c-.55 0-1.019-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 0 1-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 0 1-1.369-.49l-1.297-2.247a1.125 1.125 0 0 1 .26-1.431l1.004-.827c.292-.24.437-.613.43-.991a6.932 6.932 0 0 1 0-.255c.007-.38-.138-.751-.43-.992l-1.004-.827a1.125 1.125 0 0 1-.26-1.43l1.297-2.247a1.125 1.125 0 0 1 1.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.086.22-.128.332-.183.582-.495.644-.869l.214-1.28Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
            <button
              aria-label="Nova Senha"
              class="p-4 rounded-xl hover:bg-default-100"
            >
              <svg
                fill="none"
                height="32"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8.00001C24.0609 8.00001 25.0783 8.42143 25.8284 9.17158C26.5786 9.92172 27 10.9391 27 12M31 12C31.0002 13.1668 30.7451 14.3196 30.2527 15.3774C29.7603 16.4352 29.0424 17.3725 28.1494 18.1236C27.2565 18.8746 26.21 19.4213 25.0835 19.7251C23.9569 20.029 22.7775 20.0828 21.628 19.8827C20.8773 19.7533 20.0827 19.9173 19.544 20.456L16 24H13V27H10V30H5V26.2427C5 25.4467 5.316 24.6827 5.87867 24.1213L14.544 15.456C15.0827 14.9173 15.2467 14.1227 15.1173 13.372C14.9283 12.2802 14.968 11.1611 15.2342 10.0855C15.5003 9.00988 15.987 8.0013 16.6633 7.12364C17.3397 6.24598 18.1909 5.5184 19.1632 4.98699C20.1355 4.45558 21.2076 4.13194 22.3115 4.03658C23.4154 3.94122 24.5271 4.07622 25.5761 4.43302C26.6251 4.78982 27.5886 5.36065 28.4054 6.10931C29.2223 6.85798 29.8747 7.76817 30.3213 8.7822C30.7679 9.79623 30.999 10.892 31 12Z"
                  stroke="#A1A1AA"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M6.75 0.75C6.75 0.551088 6.67098 0.360322 6.53033 0.21967C6.38968 0.0790175 6.19891 0 6 0C5.80109 0 5.61032 0.0790175 5.46967 0.21967C5.32902 0.360322 5.25 0.551088 5.25 0.75V5.25H0.75C0.551088 5.25 0.360322 5.32902 0.21967 5.46967C0.0790175 5.61032 0 5.80109 0 6C0 6.19891 0.0790175 6.38968 0.21967 6.53033C0.360322 6.67098 0.551088 6.75 0.75 6.75H5.25V11.25C5.25 11.4489 5.32902 11.6397 5.46967 11.7803C5.61032 11.921 5.80109 12 6 12C6.19891 12 6.38968 11.921 6.53033 11.7803C6.67098 11.6397 6.75 11.4489 6.75 11.25V6.75H11.25C11.4489 6.75 11.6397 6.67098 11.7803 6.53033C11.921 6.38968 12 6.19891 12 6C12 5.80109 11.921 5.61032 11.7803 5.46967C11.6397 5.32902 11.4489 5.25 11.25 5.25H6.75V0.75Z"
                  fill="#A1A1AA"
                />
              </svg>
            </button>
          </nav>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <main
        class="h-[calc(100dvh)] grid grid-rows-[1fr_auto]"
        style="font-family: fontFamily;"
      >
        test
        <nav
          class="flex bg-white p-1 justify-center border-t-2 border-default-200"
        >
          <a
            class="p-4 rounded-xl "
            href="/"
          >
            <svg
              aria-hidden="true"
              class="h-8 text-default-400"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </a>
          <a
            class="p-4 rounded-xl hover:bg-default-100"
            href="/achivements"
          >
            <svg
              aria-hidden="true"
              class="h-8 text-default-400"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 11.25v8.25a1.5 1.5 0 0 1-1.5 1.5H5.25a1.5 1.5 0 0 1-1.5-1.5v-8.25M12 4.875A2.625 2.625 0 1 0 9.375 7.5H12m0-2.625V7.5m0-2.625A2.625 2.625 0 1 1 14.625 7.5H12m0 0V21m-8.625-9.75h18c.621 0 1.125-.504 1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125h-18c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </a>
          <a
            class="p-4 rounded-xl hover:bg-default-100"
            href="/settings"
          >
            <svg
              aria-hidden="true"
              class="h-8 text-default-400"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.325.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 0 1 1.37.49l1.296 2.247a1.125 1.125 0 0 1-.26 1.431l-1.003.827c-.293.241-.438.613-.43.992a7.723 7.723 0 0 1 0 .255c-.008.378.137.75.43.991l1.004.827c.424.35.534.955.26 1.43l-1.298 2.247a1.125 1.125 0 0 1-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.47 6.47 0 0 1-.22.128c-.331.183-.581.495-.644.869l-.213 1.281c-.09.543-.56.94-1.11.94h-2.594c-.55 0-1.019-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 0 1-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 0 1-1.369-.49l-1.297-2.247a1.125 1.125 0 0 1 .26-1.431l1.004-.827c.292-.24.437-.613.43-.991a6.932 6.932 0 0 1 0-.255c.007-.38-.138-.751-.43-.992l-1.004-.827a1.125 1.125 0 0 1-.26-1.43l1.297-2.247a1.125 1.125 0 0 1 1.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.086.22-.128.332-.183.582-.495.644-.869l.214-1.28Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </a>
          <button
            aria-label="Nova Senha"
            class="p-4 rounded-xl hover:bg-default-100"
          >
            <svg
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 8.00001C24.0609 8.00001 25.0783 8.42143 25.8284 9.17158C26.5786 9.92172 27 10.9391 27 12M31 12C31.0002 13.1668 30.7451 14.3196 30.2527 15.3774C29.7603 16.4352 29.0424 17.3725 28.1494 18.1236C27.2565 18.8746 26.21 19.4213 25.0835 19.7251C23.9569 20.029 22.7775 20.0828 21.628 19.8827C20.8773 19.7533 20.0827 19.9173 19.544 20.456L16 24H13V27H10V30H5V26.2427C5 25.4467 5.316 24.6827 5.87867 24.1213L14.544 15.456C15.0827 14.9173 15.2467 14.1227 15.1173 13.372C14.9283 12.2802 14.968 11.1611 15.2342 10.0855C15.5003 9.00988 15.987 8.0013 16.6633 7.12364C17.3397 6.24598 18.1909 5.5184 19.1632 4.98699C20.1355 4.45558 21.2076 4.13194 22.3115 4.03658C23.4154 3.94122 24.5271 4.07622 25.5761 4.43302C26.6251 4.78982 27.5886 5.36065 28.4054 6.10931C29.2223 6.85798 29.8747 7.76817 30.3213 8.7822C30.7679 9.79623 30.999 10.892 31 12Z"
                stroke="#A1A1AA"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M6.75 0.75C6.75 0.551088 6.67098 0.360322 6.53033 0.21967C6.38968 0.0790175 6.19891 0 6 0C5.80109 0 5.61032 0.0790175 5.46967 0.21967C5.32902 0.360322 5.25 0.551088 5.25 0.75V5.25H0.75C0.551088 5.25 0.360322 5.32902 0.21967 5.46967C0.0790175 5.61032 0 5.80109 0 6C0 6.19891 0.0790175 6.38968 0.21967 6.53033C0.360322 6.67098 0.551088 6.75 0.75 6.75H5.25V11.25C5.25 11.4489 5.32902 11.6397 5.46967 11.7803C5.61032 11.921 5.80109 12 6 12C6.19891 12 6.38968 11.921 6.53033 11.7803C6.67098 11.6397 6.75 11.4489 6.75 11.25V6.75H11.25C11.4489 6.75 11.6397 6.67098 11.7803 6.53033C11.921 6.38968 12 6.19891 12 6C12 5.80109 11.921 5.61032 11.7803 5.46967C11.6397 5.32902 11.4489 5.25 11.25 5.25H6.75V0.75Z"
                fill="#A1A1AA"
              />
            </svg>
          </button>
        </nav>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
